### ISSUE TITLE
Not prompted to do `stack docker pull`

### ISSUE THREAD

**Issue by ivan-m** [commercialhaskell/stack#3328]
### General summary/comments (optional)

The [documentation](https://docs.haskellstack.org/en/stable/docker_integration/#use-stack-as-normal) states that:

> The first time you run a command with a new image, you will be prompted to run `stack docker pull` to pull the image first.

However, instead I received an error message.

### Steps to reproduce

I don't know if this is relevant, but due to corporate policies I have to access Docker Hub via a mirror; I shall refer to this as `corp.mirror/remote`.

I have this in my `stack.yaml`:

```yaml
resolver: lts-8.5

docker:
  enable: false
  repo: corp.mirror/remote/fpco/stack-build
  run-args:
    - "--net=bridge"
```

```
$ stack build --docker
Running /usr/local/bin/docker inspect corp.mirror/remote/fpco/stack-build:lts-8.5 exited with ExitFailure 1

[]

Error: No such object: corp.mirror/remote/fpco/stack-build:lts-8.5
```

### Expected

`stack build --docker` should tell me I need to do a `stack docker pull`.

### Actual

```
$ stack build --docker --verbose

# Snipping stack versioning output

2017-08-09 13:23:26.609217: [debug] Checking for project config at: /project/path/stack.yaml
@(Stack/Config.hs:974:9)
2017-08-09 13:23:26.609866: [debug] Loading project config file stack.yaml
@(Stack/Config.hs:999:13)
2017-08-09 13:23:26.612122: [debug] Run process: /usr/local/bin/docker --version
@(System/Process/Read.hs:306:3)
2017-08-09 13:23:26.627167: [debug] Process finished in 14ms: /usr/local/bin/docker --version
@(System/Process/Read.hs:306:3)
2017-08-09 13:23:26.627709: [debug] Run process: /usr/local/bin/docker inspect corp.mirror/remote/fpco/stack-build:lts-8.5
@(System/Process/Read.hs:306:3)
Running /usr/local/bin/docker inspect corp.mirror/remote/fpco/stack-build:lts-8.5 exited with ExitFailure 1

[]

Error: No such object: corp.mirror/remote/fpco/stack-build:lts-8.5
```

(`stack docker pull` does indeed work.)

### Stack version

```
$ stack --version
Version 1.5.0 x86_64
```

### Method of installation

* Using nix on macOS.

**Comment by ivan-m** [comment]
I'm using Docker for Mac 17.0.6.0 with this.

I think the issue is [here](https://github.com/commercialhaskell/stack/blob/98c545266ed391107fb794d1d2969f00810498fe/src/Stack/Docker.hs#L655): if I run `docker inspect` manually:

```
$ /usr/local/bin/docker inspect corp.mirror/remote/fpco/stack-build:lts-8.5
[]
Error: No such object: corp.mirror/remote/fpco/stack-build:lts-8.5
```

There is now an extra `[]\n` more than expected by `missingImagePrefixes`.  This may be due to a change in behaviour from Docker.

**Comment by borsboom** [comment]
It should be a pretty easy fix... unfortunately parsing CLI output is going to tend toward this kind of breakage (but the Docker API is even more prone to breakage, so I don't know if using that directly would be an improvement).

Aside: since `--install-ghc` is now default behaviour, an analogous change would be to make `--docker-auto-pull` enabled by default.  I've opened #3332 to track that change.

**Comment by snoyberg** [comment]
This was addressed by #4604
